platform/src/components/aws/bucket.ts

```diff
@@ -99,6 +99,63 @@ interface BucketCorsArgs {
   maxAge?: Input<Duration>;
 }
 
+interface BucketLifecycleBaseArgs {
+  /**
+   * An S3 object key prefix that the lifecycle rule applies to.
+   * @example
+   * Applies to all the objects in the `images/` folder.
+   * ```js
+   * {
+   *   prefix: "images/"
+   * }
+   * ```
+   */
+  prefix?: Input<string>;
+  /**
+   * Whether the lifecycle rule is enabled.
+   * @example
+   * ```js
+   * {
+   *  enabled: true
+   * }
+   * ```
+   * @default `true`
+   */
+  enabled?: Input<boolean>;
+}
+
+interface BucketLifecycleExpiresInArgs extends BucketLifecycleBaseArgs {
+  /**
+   * Time after which the objects in the bucket should expire.
+   * @example
+   * ```js
+   * {
+   *  expiresIn: "30 days"
+   * }
+   * ```
+   */
+  expiresIn: Input<Duration>;
+  expiresAt?: never;
+}
+
+interface BucketLifecycleExpiresAtArgs extends BucketLifecycleBaseArgs {
+  /**
+   * Date after which the objects in the bucket should expire.
+   * @example
+   * ```js
+   * {
+   *  expiresAt: "2023-08-22"
+   * }
+   * ```
+   */
+  expiresAt: Input<string>;
+  expiresIn?: never;
+}
+
+type BucketLifecycleArgs =
+  | BucketLifecycleExpiresInArgs
+  | BucketLifecycleExpiresAtArgs;
+
```

Comment:
```suggestion
interface BucketLifecycleArgs {
  /**
   * An S3 object key prefix that the lifecycle rule applies to.
   * @example
   * Applies to all the objects in the `images/` folder.
   * ```js
   * {
   *   prefix: "images/"
   * }
   * ```
   */
  prefix?: Input<string>;
  /**
   * Whether the lifecycle rule is enabled.
   * @example
   * ```js
   * {
   *  enabled: true
   * }
   * ```
   * @default `true`
   */
  enabled?: Input<boolean>;
  /**
   * Time after which the objects in the bucket should expire.
   * @example
   * ```js
   * {
   *  expiresIn: "30 days"
   * }
   * ```
   */
  expiresIn?: Input<`${number} days` | `${number} day`>;
  /**
   * Date after which the objects in the bucket should expire.
   * @example
   * ```js
   * {
   *  expiresAt: "2023-08-22"
   * }
   * ```
   */
  expiresAt?: Input<string>;
}

```

i had trouble generating the docs with the type union and the never  

less correct but simpler  

also, switched Duration to `${number} days` because the API only takes days at minimum. even better if you add `DurationDays` to `duration.ts` alongside the other specific durations

Reply:
Added Duration days and will remove union type (sad)